### DISCUSSION TITLE
Use handleChange outside Formik

### DISCUSSION THREAD

**Discussion by cnfreitax** [jaredpalmer/formik#3288] [General]
Hi! I needed make a dynamic form, for this I used Form render and FieldArray. But, I wanted use handleChange method from Formik outside form tto deal manipulation of data. 

Have way to this?

**Comment by johnrom** [comment]
Yes, you can pass the prop `innerRef` to Formik, and use that _after mount_ in order to update data. Note that formikRef.current will be undefined before the component has mounted.

```
const formikRef = React.useRef();

// Formik will have mounted and instantiated the ref by the time the event handlers can be triggered
return <>
  <Formik innerRef={formikRef} initialValues={{ myField: '' }} />
  <input name="myField" onChange={formikRef.current?.handleChange} onBlur={formikRef.current?.handleBlur} />
</>
```
